### PR TITLE
Improve code readability and test suite

### LINKED ISSUE #115: Code Clean Up
Code clean up for the company class 

specifically
1. all the find commands
2. delete person from company
3. add person to company

### PR DESCRIPTION
**Pull request by ryantzr1**
Fixes #115 

Changes made:
1. removed duplicate test cases
2. removed dead code
3. make code more readable
4. improve test suite
5. add some javadocs
6. renamed some variables

### CODE REVIEW

File: src/main/java/connectify/logic/commands/AddPersonCommand.java [hunk 1]
```diff
@@ -64,16 +64,9 @@ public AddPersonCommand(Person person, Index companyIndex) {
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
 
-        if (model.hasPerson(toAdd)) {
-            throw new CommandException(MESSAGE_DUPLICATE_PERSON);
-        }
+        validateAddition(model);
```
Comment (nreHieW) [inline src/main/java/connectify/logic/commands/AddPersonCommand.java:67]: Thanks for spotting this!

File: src/main/java/connectify/logic/commands/AddPersonCommand.java [hunk 2]
```diff
@@ -64,16 +64,9 @@ public AddPersonCommand(Person person, Index companyIndex) {
...
 
         List<Company> lastShownList = model.getFilteredCompanyList();
-
-        if (companyIndex.getZeroBased() >= lastShownList.size()) {
-            throw new CommandException(Messages.MESSAGE_INVALID_COMPANY_DISPLAYED_INDEX);
-        }
-
```
Comment (nreHieW) [inline src/main/java/connectify/logic/commands/AddPersonCommand.java:76]: Thanks for spotting this!

**Review by nreHieW — APPROVED** [review]
LGTM 👍

File: src/main/java/connectify/logic/commands/AddPersonCommand.java [hunk 3]
```diff
@@ -85,6 +78,27 @@ public CommandResult execute(Model model) throws CommandException {
...
+    private void validateAddition(Model model) throws CommandException {
+        if (model.hasPerson(toAdd)) {
+            throw new CommandException(MESSAGE_DUPLICATE_PERSON);
+        }
+
+        List<Company> lastShownList = model.getFilteredCompanyList();
+        if (companyIndex.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_COMPANY_DISPLAYED_INDEX);
+        }
+    }
```
Comment (tayian) [inline src/main/java/connectify/logic/commands/AddPersonCommand.java:99]: this looks to be a powerful validation check, thanks for including it!

**Pull request merged**